pkg/bootstrap/cni_test.go

```diff
@@ -85,18 +86,21 @@ func TestCNISupportedVersionRendering(t *testing.T) {
 					assert.Success(err, "resource decoding", t)
 
 					containers, found, err := unstructured.NestedSlice(resource.UnstructuredContent(), "spec", "template", "spec", "containers")
+					if tc.name == "v2.3 only" && !found {
+						// moved v2.3 cni container into a new daemonset, skip testing the old one
+						continue
+					}
```

Comment:
Because in v2.3 case, we will have two daemonsets. One from our original daemonset.yaml and one from the new template daemonset-v2.3.yaml.

When we create a v2.3 SMCP, supportedVersion only has [v2.3] value, so the daemonset.yaml will not include any container. That's why 
```
unstructured.NestedSlice(resource.UnstructuredContent(), "spec", "template", "spec", "containers") 
```
returns nil and found is false in that case.

However, if I change the rest of the if statements in the daemonset.yaml, that will break existing clusters in upgrade.
In that case, only removing the v2.3 container section in the daemonset.yaml template will be safe in upgrade.

Reply:
Ok, thanks! Now I understand. Could you add more precise comment?

Reply:
sure, i will more